### DISCUSSION TITLE
Set localePrefix prop based on domains

### DISCUSSION THREAD

**Discussion by markoleavy** [amannn/next-intl#1030] [Q&A] [answered]
Hi!
I'd like to be able to set optional locale based on different domains, in the same way we define the default locale. Example: ".it" is just for Italian, and does not show any locale, '.com' is for the other languages and must always show a locale.
Ideally, the middleware config should be like that:
```js
export default createMiddleware({
  locales: ['en-US', 'fr-FR', 'en-GB', 'de-DE', 'it-IT']
  domains: [
          {
            domain: 'www.mywebsite.it',
            defaultLocale: 'it-IT',
            localePrefix: 'as-needed',
            locales: ['it-IT'],
          },
          {
            domain: 'www.mywebsite.com',
            defaultLocale: 'en-US',
            localePrefix: 'always',
            locales: ['en-US', 'fr-FR', 'en-GB', 'de-DE'],
          },
        ],
  localeDetection: true,
});
```
Is there a way to achieve this?

**Comment by markoleavy** [comment]
I managed to get this result using `defaultLocale = ''` for 'www.mywebsite.com'. 
This also gives me access to the `page.ts` file outside the `[locale]` folder on .com, where I'd like to put the code for a language-disambiguation page.

@amannn is that the supposed behaviour  or is it just an 'involuntary-feature' that might change unpredictably with updates?

**Reply by amannn** [reply, marked as answer]
Hey, thanks for the question! Currently this is undocumented behavior, I'm not sure what consequences this has. And yes, it could break with updates.

You can open a feature request though if this use case is relevant for you!
